### PR TITLE
extract unison-hashing-v2 package from unison-parser-typechecker

### PR DESCRIPTION
**Pull request by mitchellwrosen**
## Overview

This PR extracts a `unison-hashing-v2` package out of `unison-parser-typechecker`. Working towards performing hashing at the database level (to support updating type / type mentions indices from share pull code path(s)), which would otherwise cause a circular dependency. (Also, working towards breaking `parser-typechecker` up into smaller components).

### CODE REVIEW

File: hie.yaml [hunk 1]
```diff
@@ -1,74 +1,95 @@
 cradle:
   stack:
-    - path: "codebase2/codebase"
+    - path: "codebase2/codebase/./"
```
Comment (ChrisPenner) [inline hie.yaml:3]: What's going on here?

File: hie.yaml [hunk 2]
```diff
@@ -1,74 +1,95 @@
...
 
-    - path: "unison-cli/transcripts"
+    - path: "unison-cli/transcripts/Transcripts.hs"
       component: "unison-cli:exe:transcripts"
 
-    - path: "unison-cli/unison"
+    - path: "unison-cli/unison/Main.hs"
       component: "unison-cli:exe:unison"
 
+    - path: "unison-cli/unison/ArgParse.hs"
```
Comment (ChrisPenner) [inline hie.yaml:72]: Do path prefixes not work in `exe`'s? I know my hls doesn't usually work for executables, but curious if this is the fix 😄

File: hie.yaml [hunk 3]
```diff
@@ -1,74 +1,95 @@
...
 
-    - path: "unison-share-api/src"
-      component: "unison-share-api:lib"
-
     - path: "lib/unison-util-relation/src"
       component: "unison-util-relation:lib"
 
     - path: "lib/unison-util-relation/test"
-      component: "unison-util-relation:test:tests"
+      component: "unison-util-relation:test:util-relation-tests"
```
Comment (ChrisPenner) [inline hie.yaml:46]: I don't see where you re-name the appropriate test-suite in this PR; or was this just incorrectly named in the first place?

**Review by ChrisPenner — APPROVED** [review]
Not sure what's going on with the `hie.yaml`; but the idea of splitting off hashing sounds good 👍🏼

**Comment by mitchellwrosen** [comment]
@ChrisPenner All changes to `hie.yaml` are due to:

```
gen-hie > hie.yaml
```

File: hie.yaml (hunk 3 above)
Comment (mitchellwrosen) [reply hie.yaml:46]: The latter, it seems

File: hie.yaml (hunk 2 above)
Comment (mitchellwrosen) [reply hie.yaml:72]: No clue... I'm torn between trusting `gen-hie`'s output and being suspicious of it. I guess I can try loading these files into hls and seeing if it barfs.

**Pull request merged**